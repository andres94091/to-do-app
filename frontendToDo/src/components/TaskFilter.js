import React from 'react'
import '../index.css';

const TaskFilter = ({setFilterStatus}) => {

  const handleFilterInput = (event) => {
    setFilterStatus(event.target.value)
  }

  return (
    <>
      <form className='filter-todos' onChange={handleFilterInput}>
        <input className="form-check-input" type="radio" id='null' name='filer' value="null" />
        <label className="form-check-label" for="null">
          All
          </label>
        <input className="form-check-input" type="radio" id='true' name='filer' value="true" />
        <label className="form-check-label" for="true">
          Done
          </label>
        <input className="form-check-input" type="radio" id='false' name='filer' value="false" />
        <label className="form-check-label" for="false">
          NotDone
          </label>
      </form>

    </>
  )
}

export default TaskFilter
