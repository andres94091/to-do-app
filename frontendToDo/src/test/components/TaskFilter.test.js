import '@testing-library/jest-dom';

import React from 'react';
import { shallow } from 'enzyme';
import TaskFilter from '../../components/TaskFilter';

const setFilterStatus = jest.fn()

describe('tests in <TaskFilter/>', () => {

  const wrapper = shallow(<TaskFilter
    setFilterStatus= {setFilterStatus}
  />)

  it('should render the component correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('should change the filter status for tasks', () => {

    wrapper.find('form').simulate('change', {target:{value:'null'}})

    expect(setFilterStatus).toHaveBeenCalled()
  })
  
  
  
  
})






