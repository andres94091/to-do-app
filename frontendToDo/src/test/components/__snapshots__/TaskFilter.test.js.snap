// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests in <TaskFilter/> should render the component correctly 1`] = `
<Fragment>
  <form
    className="filter-todos"
    onChange={[Function]}
  >
    <input
      className="form-check-input"
      id="null"
      name="filer"
      type="radio"
      value="null"
    />
    <label
      className="form-check-label"
      for="null"
    >
      All
    </label>
    <input
      className="form-check-input"
      id="true"
      name="filer"
      type="radio"
      value="true"
    />
    <label
      className="form-check-label"
      for="true"
    >
      Done
    </label>
    <input
      className="form-check-input"
      id="false"
      name="filer"
      type="radio"
      value="false"
    />
    <label
      className="form-check-label"
      for="false"
    >
      NotDone
    </label>
  </form>
</Fragment>
`;
